src/ngraph/shape_util.hpp

```diff
@@ -56,9 +57,50 @@ namespace ngraph
         return project(axis_values, axes);
     }
 
+    template<class I,class V>
```

Comment:
Naming things I and V and not putting any doc strings makes this unfathomable. What is this doing?